OpenSim/Simulation/Model/Model.cpp

```diff
@@ -656,38 +666,31 @@ void Model::extendConnectToModel(Model &model)
             Joint* free = new FreeJoint(jname, *ground, *child);
             free->upd_reverse() = mob.isReversedFromJoint();
             addJoint(free);
+            setNextSubcomponentInSystem(*free);
```

Comment:
I was expecting this code to be more similar to the code above (lines 645–650) when a WeldConstraint is added. The comment above about JointSet (lines 611–612) suggests that JointSet will eventually disappear, so should `addJoint(free)` be `adoptSubcomponent(free)` instead? (In `addJoint()`, it looks like `free` will be adopted by the JointSet, which will eventually disappear.)